### ISSUE TITLE
Vanishing gradient in attention-based Graph Neural Network

### ISSUE THREAD

**Issue by richardrl** [rusty1s/pytorch_geometric#215]
I have implemented a simple attention-based graph neural network. It is simply stacked depth-wise relational modules from Santoro 17. 

The problem is, my gradients are quickly vanishing in my sparse label setting. I am already using residual connections and layer normalization. What graph architectures here might address this issue?

**Comment by richardrl** [comment]
I am trying to replicate the architecture here, described in page 3, but with additive attention: https://openreview.net/pdf?id=HkxaFoC9KQ

Even with 2 'relational' modules, the gradients seem to vanish. Typically these attentional graph neural network architectures use lots of dense layers so I'm curious how any 'deeper' graph networks could even be trained if my "2 attention deep" architecture is vanishing

**Comment by rusty1s** [comment]
Are you sure there is no bug in your implementation? How is your operator implemented?

**Comment by rusty1s** [comment]
I am closing this for now. Feel free to reopen in case you want to discuss this further.
